# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/rect.glsl
# Commit: c0f58fd74e56
# Full Hash: c0f58fd74e56b24f67fccae08097d151e21a0f80
# Author: Ashley Hale <ahale@mozilla.com>
# Date: 2024-05-09 16:15:25
# Regressor Bug: 1824502
# File Overlap Count: 1
# Description:
#   Bug 1824502 - Implement native SVG filter rendering in WebRender r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D209646
# ==============================================================================

diff -r 3e2ec3a3b738 -r c0f58fd74e56 gfx/wr/webrender/res/rect.glsl
--- a/gfx/wr/webrender/res/rect.glsl	Thu May 09 02:12:39 2024 +0000
+++ b/gfx/wr/webrender/res/rect.glsl	Thu May 09 04:09:00 2024 +0000
@@ -38,3 +38,12 @@
 vec2 rect_size(RectWithEndpoint rect) {
     return rect.p1 - rect.p0;
 }
+
+// this is similar to rect_clamp but repeats the image for coordinates outside
+// the rect, used in SVG feTile filter
+vec2 rect_repeat(vec2 p, vec2 p0, vec2 p1) {
+    vec2 r = p - p0;
+    vec2 s = p1 - p0;
+    vec2 is = 1.0 / max(s, vec2(0.000001));
+    return p0 + s * fract(is * r);
+}